docs/griptape-framework/structures/tasks.md

```diff
@@ -146,3 +146,254 @@ agent.run()
+John, 25, 123 Main St
+Jane, 30, 456 Elm St
+"""
+
+columns = ["Name", "Age", "Address"]
+
+
+# Create an agent and add the ExtractionTask to it
+agent = Agent()
+agent.add_task(
+    ExtractionTask(
+        input_template="{{ csv_data }}",
+        context={"csv_data": csv_data},
+        extraction_engine=csv_extraction_engine,
+        args={"column_names": columns},
+    )
+)
+
+print("CSV extraction task.")
+# Run the agent
+agent.run()
+
+json_extraction_engine = JsonExtractionEngine()
+
+# Define some JSON data
+json_data = """
+[
+  {"Name": "John", "Age": "25", "Address": "123 Main St"},
+  {"Name": "Jane", "Age": "30", "Address": "456 Elm St"}
+]
+"""
+
+# Add the JsonExtraction task to the same agent
+agent.add_task(
+    ExtractionTask(
+        input_template="{{ json_data }}",
+        context={"json_data": json_data},
+        extraction_engine=json_extraction_engine,
+        args={"template_schema": {"Name": "{{ name }}", "Age": "{{ age }}"}},
+    )
+)
+
+print("JSON extraction task.")
+# Run the agent
+agent.run()
+```
+```
+CSV extraction task.
+[10/12/23 17:13:54] INFO     ExtractionTask d7a03ff4f7944bad8755dfd43fd3bd66    
+                             Input:                                             
+                             Name, Age, Address                                 
+                             John, 25, 123 Main St                              
+                             Jane, 30, 456 Elm St                               
+                                                                                
+[10/12/23 17:13:55] INFO     ExtractionTask d7a03ff4f7944bad8755dfd43fd3bd66    
+                             Output: John,"""25""","""123 Main St"""            
+                             Jane,"""30""","""456 Elm St"""                     
+JSON extraction task.
+                    INFO     ExtractionTask eee005faa0f043bea738a29cb413036d    
+                             Input:                                             
+                             [                                                  
+                               {"Name": "John", "Age": "25", "Address": "123    
+                             Main St"},                                         
+                               {"Name": "Jane", "Age": "30", "Address": "456 Elm
+                             St"}                                               
+                             ]                                                  
+                                                                                
+[10/12/23 17:13:56] INFO     ExtractionTask eee005faa0f043bea738a29cb413036d    
+                             Output: {'name': 'John', 'age': '25'}              
+                             {'name': 'Jane', 'age': '30'}     
+```
+
+
+# Text Summary Task
+
+To summarize a text, you can use the TextSummaryTask. This task allows us to pass in a Summarization Engine and a set of arguments to the engine.
```

Comment:
Link to the reference docs for Summarization Engine.

Reply:
Done~

Reply:
Looks like you're now linking to this page instead of the reference docs. Search around the docs for `reference` to see how to link to there instead.